library/modules/Textures.cpp

```diff
@@ -292,13 +412,33 @@ static void uninstall_reset_point() {
     INTERPOSE_HOOK(tracking_stage_new_arena, logic).remove();
 }
 
+static void reserve_static_range() {
+    if (static_cast<size_t>(enabler->textures.init_texture_size) != enabler->textures.raws.size()) {
+        WARN(textures).print(
+            "reserved range can't be installed! all textures will be loaded to dynamic range!");
+        return;
+    }
+    reserved_range.init(enabler->textures.init_texture_size);
+    auto dummy_surface =
+        DFSDL_CreateRGBSurfaceWithFormat(0, 0, 0, 32, SDL_PixelFormatEnum::SDL_PIXELFORMAT_RGBA32);
+    for (int32_t i = 0; i < ReservedRange::size; i++) {
+        add_texture(dummy_surface);
```

Comment:
at program close, DF will attempt to free all these textures. Is the refcount incremented somewhere so that it doesn't end up doing a double free?

Reply:
game will perform `FreeSurface` for entire `textures.raws` (i guess)... so every texture with `refcount = 1` will be destroyed (here we destroy dummy)
our cached surfaces has `refcount = 2` and it will be decrmented to 1..
then `Texture::cleanup()` will be called in `Core` shutdown sequence, there we will one more time call `FreeSurface` for our cached surfaces and finaly destroy it

Reply:
I note that we currently double free on exit with current code in 50.09-r3, so this is a real issue. I'll test to make sure this PR fixes that issue before merging, but otherwise this is looking pretty good